report/main.tex

```diff
@@ -52,18 +63,171 @@ \section{Introduction}
 We used the ART paper~\cite{paranjape2023art}.
 
 
-\begin{ack}
-  Thanks Mom.
-\end{ack}
+\section{Background}
+
+
+\section{Methodology}
+%- provide tools that are analogous to long term memory in humans
+
+Our goal was to create a modified ART framework that allows for an LLM to read from, write to, and organize a form of long-term memory, without actually retraining model weights.
+This would allow for the model to dynamically access previously stored information, as well as update its local repository with valuable data as it reads through prompts.
+Given the finite context window of any LLM, ensuring that only relevant data is retrieved would be a crucial requirement of this system.
+With these goals and constraints in mind, we chose a Python dictionary as the underlying data structure for persisting information.
+A dictionary naturally organizes data into key-value pairs, which would naturally allow for organization and efficient retrieval based on the LLM selecting only relevant keys.
+The LLM would load the information stored at keys it finds relevant to its current task, and organize the information it stores by choosing an appropriate key.
+
+In the context of ART, this translates into the creation of three tools to accomplish our envisioned task. Firstly, we define \texttt{[list-keys]}, a tool that takes no inputs and returns a string containing all the keys in the dictionary.
```

Comment:
are we explaining twice the three tools we designed?

Reply:
Idk, Alex you decide where you prefer the tools to be described